datafusion/physical-expr/src/intervals/cp_solver.rs

```diff
@@ -1414,4 +1415,130 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_propagate_comparison() {
+        let left = Interval::new(
+            IntervalBound::make_unbounded(DataType::Int64).unwrap(),
+            IntervalBound::make_unbounded(DataType::Int64).unwrap(),
+        );
+        let right = Interval::new(
+            IntervalBound::new(ScalarValue::Int64(Some(1000)), false),
+            IntervalBound::new(ScalarValue::Int64(Some(1000)), false),
+        );
+        assert_eq!(
```

Comment:
It took me some time to work this out, so I figured it might be helpful to leave some comments explaining what is expected. 

```suggestion
        // left is unbounded, right is known to be [1000,1000]
        // so left < right results in knowing that left is now < 1000 
        assert_eq!(
```